pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -0,0 +1,719 @@
+
+	// do we need oracle service?
+
+	// var oracleService *oracle.Oracle
+	//
+	// if cfg.ApplicationConfiguration.Oracle.Enabled {
+	// 	var c oracle.Config
+	// 	c.Log = log
+	// 	c.Network = cfg.ProtocolConfiguration.Magic
+	// 	c.MainCfg = cfg.ApplicationConfiguration.Oracle
+	// 	c.Chain = artifacts.blockchain
+	// 	c.OnTransaction = artifacts.netServer.RelayTxn
+	//
+	// 	oracleService, err = oracle.NewOracle(c)
+	// 	if err != nil {
+	// 		return fmt.Errorf("init Oracle module: %w", err)
+	// 	}
+	//
+	// 	artifacts.blockchain.SetOracle(oracleService)
+	// 	artifacts.netServer.AddService(oracleService)
+	// }
+
+	if isAlphabetNode {
+		// currently Consensus and Notary services doesn't support external
+		// wallet.Account, so just pass file with the wallet including
+		// signTools.Account. Password should work since the account has been already
+		// unlocked with it.
+		cfgUnlockWallet := config.Wallet{
+			Path:     signTools.AccountFile.Path,
+			Password: signTools.AccountFile.Password,
+		}
+
+		var cfgConsensus consensus.Config
+		cfgConsensus.Logger = log
+		cfgConsensus.Broadcast = netServer.BroadcastExtensible
+		cfgConsensus.Chain = blockChain
+		cfgConsensus.ProtocolConfiguration = blockChain.GetConfig().ProtocolConfiguration
+		cfgConsensus.RequestTx = netServer.RequestTx
+		cfgConsensus.StopTxFlow = netServer.StopTxFlow
+		// currently consensus.Service doesn't support external wallet.Account, so just
+		// pass file with the wallet including signTools.Account. Password should work
+		// since the account has been already unlocked with it.
+		cfgConsensus.Wallet = cfgUnlockWallet
+		cfgConsensus.TimePerBlock = cfgServer.TimePerBlock
+
+		consensusService, err := consensus.NewService(cfgConsensus)
+		if err != nil {
+			return nil, fmt.Errorf("init Consensus module: %w", err)
+		}
+
+		netServer.AddConsensusService(consensusService, consensusService.OnPayload, consensusService.OnTransaction)
+
+		var cfgNotary notarysvc.Config
+		cfgNotary.MainCfg = config.P2PNotary{
+			Enabled:      true,
+			UnlockWallet: cfgUnlockWallet,
+		}
+		cfgNotary.Chain = blockChain
+		cfgNotary.Log = log
+
+		notaryService, err := notarysvc.NewNotary(cfgNotary, netServer.Net, netServer.GetNotaryPool(), func(tx *transaction.Transaction) error {
+			err := netServer.RelayTxn(tx)
+			if err != nil && !errors.Is(err, core.ErrAlreadyExists) {
+				return fmt.Errorf("relay completed notary transaction %s: %w", tx.Hash().StringLE(), err)
+			}
+
+			return nil
+		})
+		if err != nil {
+			return nil, fmt.Errorf("init Notary service: %w", err)
+		}
+
+		netServer.AddService(notaryService)
+		blockChain.SetNotary(notaryService)
+	}
+
+	// "make" channel rw to satisfy Start method
+	chErrRw := make(chan error)
+
+	go func() {
+		err, ok := <-chErrRw
+		if ok {
+			chErr <- err
+		}
+	}()
+
+	rpcServer := rpcsrv.New(blockChain, cfg.ApplicationConfiguration.RPC, netServer, nil, log, chErrRw)
+
+	netServer.AddService(&rpcServer) // is it a cyclic dependency?
```

Comment:
Not exactly, but RPC needs a network server and then network server can control RPC as any other service (it's an interface here).